### PR TITLE
Mask Padded Tokens In Aspect Ratio Mask Preparation

### PR DESCRIPTION
**Pull request by vamsi-kamanuru**
# What does this PR do?

Problem: The function `_prepare_aspect_ratio_attention_mask` is not correctly masking padded tokens. The output currently assigns 0s to both padded and non-padded tokens in the vision encoder.

Solution: Update `_prepare_aspect_ratio_attention_mask` to correctly mask padded tokens by setting their mask values to a large negative number.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by vamsi-kamanuru** [comment]
@ArthurZucker, @amyeroberts, @qubvel Please take a look, thank you!!

**Comment by vamsi-kamanuru** [comment]
@ArthurZucker @zucchini-nlp A gentle reminder. I don't seem to have the permissions to add any reviewers.

**Review by ArthurZucker — COMMENTED** [review]
Sorry for the delay! 
Could you add a test? In general this is the safest way to make sure we indeed fixed the issue!
